### DISCUSSION TITLE
configure.pm refuses to enable-fips

### DISCUSSION THREAD

**Discussion by gm2518** [openssl/openssl#23332] [Q&A] [answered]
Hi..
Have successfully compiled and installed openssl v 3.2.0 on Win 11 x64 environment. However, even though during compilation, 
(a) -enable-fips was specified, and 
(b) fipsmodule.cnf was produced, and
(c) it was included in the openssl.cnf,
the acvp software declared that the fips test at the startup is failing.
A sample output is reproduced here
```
Crypto module returned failure code when running quick digest.
Error occured when testing FIPS at startup (rv = 10). Please verify the FIPS provider is
properly installed and configured. Exiting...
```
Then on some more investigation, I realized that the even though -enable-fips is being specified in the command line, the command perl configdata.pm --dump shows the following 
(am showing just the truncated part, the complete dump is attached
```
Command line (with current working directory = .):

C:\Users\gautam\Perl\perl\bin\perl.exe Configure -enable-fips -enable-acvp-tests -no-deprecated --prefix=c:\users\gautam\openssl_fips --openssldir=c:\users\gautam\openssl_fips VC-WIN64A
Perl information:
    C:\Users\gautam\Perl\perl\bin\perl.exe
    5.38.0 for MSWin32-x64-multi-thread
Enabled features:
...

Disabled features:

    acvp-tests          [cascade] OPENSSL_NO_ACVP_TESTS
    deprecated          [option]  OPENSSL_NO_DEPRECATED
    dynamic-engine      [cascade] 
    ec_nistp_64_gcc_128 [default] OPENSSL_NO_EC_NISTP_64_GCC_128
    fips                [default] 
    fips-securitychecks [cascade] OPENSSL_NO_FIPS_SECURITYCHECKS

Config target attributes:
...
```

How do I ensure that -enable-fips is enabled ?
I have the same behavior in ver 3.0.8 (the FIPS certified version). My end objective is to compile 3.0.8, and transplant the fips module to 3.2.0.

Thanks for your attention and time

[opssl_issue.txt](https://github.com/openssl/openssl/files/13976271/opssl_issue.txt)

**Comment by nhorman** [comment, marked as answer]
features passed to Configure don't accept leading '-' tokens, as those indicate additional cflags/options.  Try configuring like this:
```
C:\Users\gautam\Perl\perl\bin\perl.exe Configure enable-fips enable-acvp-tests -no-deprecated --prefix=c:\users\gautam\openssl_fips --openssldir=c:\users\gautam\openssl_fips VC-WIN64A
```
